x/stake/types/validator_test.go

```diff
@@ -274,3 +274,76 @@ func TestValidatorMarshalUnmarshalJSON(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, validator, *got)
 }
+
+func TestValidatorSetInitialCommission(t *testing.T) {
+	val := NewValidator(addr1, pk1, Description{})
+	testCases := []struct {
+		validator   Validator
+		commission  Commission
+		blockTime   time.Time
+		expectedErr bool
+	}{
+		{
+			validator:   val,
+			commission:  NewCommission(sdk.ZeroDec(), sdk.ZeroDec(), sdk.ZeroDec()),
+			blockTime:   time.Now().UTC(),
+			expectedErr: false,
+		},
+		{
+			validator:   val,
+			commission:  NewCommission(sdk.ZeroDec(), sdk.NewDecWithPrec(-1, 1), sdk.ZeroDec()),
+			blockTime:   time.Now().UTC(),
+			expectedErr: true,
+		},
+		{
+			validator:   val,
+			commission:  NewCommission(sdk.ZeroDec(), sdk.NewDec(15000000000), sdk.ZeroDec()),
+			blockTime:   time.Now().UTC(),
+			expectedErr: true,
+		},
+		{
+			validator:   val,
+			commission:  NewCommission(sdk.NewDecWithPrec(-1, 1), sdk.ZeroDec(), sdk.ZeroDec()),
+			blockTime:   time.Now().UTC(),
+			expectedErr: true,
+		},
+		{
+			validator:   val,
+			commission:  NewCommission(sdk.NewDecWithPrec(2, 1), sdk.NewDecWithPrec(1, 1), sdk.ZeroDec()),
+			blockTime:   time.Now().UTC(),
+			expectedErr: true,
+		},
+		{
+			validator:   val,
+			commission:  NewCommission(sdk.ZeroDec(), sdk.ZeroDec(), sdk.NewDecWithPrec(-1, 1)),
+			blockTime:   time.Now().UTC(),
+			expectedErr: true,
+		},
+		{
+			validator:   val,
+			commission:  NewCommission(sdk.ZeroDec(), sdk.NewDecWithPrec(1, 1), sdk.NewDecWithPrec(2, 1)),
+			blockTime:   time.Now().UTC(),
+			expectedErr: true,
```

Comment:
Same comment as above about compressed definitions during table-driven tests. Let's shorten this